benchmark-android/src/main/java/org/conscrypt/EngineType.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.conscrypt;
+
+import static org.conscrypt.TestUtils.PROTOCOL_TLS_V1_2;
+import static org.conscrypt.TestUtils.initClientSslContext;
+import static org.conscrypt.TestUtils.initEngine;
+import static org.conscrypt.TestUtils.initServerSslContext;
+
+import java.security.KeyStore.PrivateKeyEntry;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+import libcore.java.security.TestKeyStore;
+
+/**
+ * Enumeration of various types of engines for use with engine-based benchmarks.
+ */
+@SuppressWarnings({"ImmutableEnumChecker", "unused"})
+public enum EngineType {
```

Comment:
Yep, it needs to not have any references to Netty.

It could be an interface.  The main downside I can see is that it would make the base benchmark classes unable to have the main() methods.  They could be moved to the Jmh versions, though it seems a bit odd to run the Jmh class specifically not under Jmh, but not a huge deal.  I think I'm fine either way, if you have a preference for the interface version I can change it.